From e8c8c7c22e35c6c6c4933e39a3c87e0ba03711a3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Richard=20Hoz=C3=A1k?=
 <5235838+richardhozak@users.noreply.github.com>
Date: Wed, 12 Aug 2026 17:48:19 +0200
Subject: [PATCH] Fix vector return type to match what is returned

`vcleq_f32` returns `uint32x4_t`, which is also the mask type expected by
`vbslq_f32`. GCC 15 rejects the implicit conversions accepted by Clang.

Backport of https://github.com/google/filament/pull/10299.
---
 filament/src/details/ColorGradingNeon.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/filament/src/details/ColorGradingNeon.h b/filament/src/details/ColorGradingNeon.h
index 282bc1e7e81a..79b58f23bf35 100644
--- a/filament/src/details/ColorGradingNeon.h
+++ b/filament/src/details/ColorGradingNeon.h
@@ -179,9 +179,9 @@ inline float32x4_t v_atan2q_f32(float32x4_t const y, float32x4_t const x) {
  */
 UTILS_ALWAYS_INLINE
 inline void v_oetf_sRGB(float32x4_t& cg_r, float32x4_t& cg_g, float32x4_t& cg_b) {
-    float32x4_t const r_cond = vcleq_f32(cg_r, vdupq_n_f32(0.0031308f));
-    float32x4_t const g_cond = vcleq_f32(cg_g, vdupq_n_f32(0.0031308f));
-    float32x4_t const b_cond = vcleq_f32(cg_b, vdupq_n_f32(0.0031308f));
+    uint32x4_t const r_cond = vcleq_f32(cg_r, vdupq_n_f32(0.0031308f));
+    uint32x4_t const g_cond = vcleq_f32(cg_g, vdupq_n_f32(0.0031308f));
+    uint32x4_t const b_cond = vcleq_f32(cg_b, vdupq_n_f32(0.0031308f));
 
     float32x4_t const r_lin = vmulq_n_f32(cg_r, 12.92f);
     float32x4_t const g_lin = vmulq_n_f32(cg_g, 12.92f);
-- 
2.51.0
